torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -96,6 +96,23 @@ static int64_t floordiv(int64_t a, int64_t b) {
     return (r.rem) ? r.quot - 1 : r.quot;
   }
 }
+void checkDoubleInRange(double a) {
+  if (std::isnan(a) || std::isinf(a) ||
+      a > double(std::numeric_limits<int64_t>::max()) ||
+      a < double(std::numeric_limits<int64_t>::min())) {
+    throw std::runtime_error(
```

Comment:
Can you use our internal error `c10::Error`? We haven't been great about maintaining this, but in certain places (like constant propagation) we rely on  distinguishing errors thrown by us versus general c++ errors

Reply:
It seems we actually don't use `c10::Error` anywhere, except for one instance in constant prop. All the other similar errors in this file are also using `std::runtime_error`, should we eventually switch all of them?

Reply:
Yea, we should eventually switch them. Every AT_ASSERT, AT_ERROR, etc uses `c10::Error` so it's not just the one place that we catch currently

Reply:
Done.